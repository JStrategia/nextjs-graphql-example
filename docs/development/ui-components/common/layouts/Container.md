# Container

Renders a `div` tag with container style (correct width for screen).

## Usage

```tsx
function MyComponent() {
  return (
    <Container>
      <div />
    </Container>
  );
}
```
