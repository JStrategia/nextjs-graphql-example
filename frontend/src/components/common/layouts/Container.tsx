import styled, { css } from 'styled-components';

const Container = styled.div(
  ({ theme }) => css`
    max-width: ${theme.spacing.container};
    width: 100%;
    margin-left: auto;
    margin-right: auto;
    box-sizing: border-box;
    display: block;
  `,
);

export default Container;
