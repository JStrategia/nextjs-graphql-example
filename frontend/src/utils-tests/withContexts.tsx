import React from 'react';
import ThemeProvider from 'contexts/ThemeProvider';

function withContexts(children: React.ReactNode) {
  return <ThemeProvider>{children}</ThemeProvider>;
}

export default withContexts;
