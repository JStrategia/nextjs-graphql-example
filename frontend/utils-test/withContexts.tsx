import React, { ReactNode } from 'react';
import ThemeProvider from '../contexts/ThemeProvider';

function withContexts(children: ReactNode) {
  return <ThemeProvider>{children}</ThemeProvider>;
}

export default withContexts;
