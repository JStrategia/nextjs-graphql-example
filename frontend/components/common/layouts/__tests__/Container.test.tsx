import * as React from 'react';
import { create } from 'react-test-renderer';
import Container from '../Container';
import withContexts from '../../../../utils-test/withContexts';

describe('Container', () => {
  describe('Given no children', () => {
    describe('When rendering', () => {
      const render = create(withContexts(<Container />)).toJSON();

      test('Then match snapshot', () => {
        expect(render).toMatchSnapshot();
      });
    });
  });

  describe('Given children', () => {
    const children = <span>Children</span>;

    describe('When rendering', () => {
      const render = create(withContexts(<Container>{children}</Container>)).toJSON();

      test('Then match snapshot', () => {
        expect(render).toMatchSnapshot();
      });
    });
  });
});
