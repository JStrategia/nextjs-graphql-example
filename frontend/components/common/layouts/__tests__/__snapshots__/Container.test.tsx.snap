// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Given children Should should match snapshot 1`] = `
<div
  className="sc-bczRLJ eGDUch"
>
  <span>
    Children
  </span>
</div>
`;

exports[`Container Given no children Should match snapshot 1`] = `
<div
  className="sc-bczRLJ eGDUch"
/>
`;
